packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2072,7 +2072,21 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return this.finishNode(nonNullExpression, "TSNonNullExpression");
       }
 
+      let isOptionalCall = false;
+      if (
+        this.match(tt.questionDot) &&
+        this.lookaheadCharCode() === charCodes.lessThan
+      ) {
+        state.optionalChainMember = isOptionalCall = true;
```

Comment:
Nit: we can move this after the `if (noCalls) {}` check, since we don't need to set if if we stop and return.